File: android/smartfoo-android-lib-core/src/main/java/com/smartfoo/android/core/view/FooViewUtils.java
package com.smartfoo.android.core.view;

import android.view.View;

public class FooViewUtils
{
    public static String viewVisibilityToString(int visibility)
    {
        String name;
        switch (visibility)
        {
            case View.VISIBLE:
                name = "VISIBLE";
                break;
            case View.INVISIBLE:
                name = "INVISIBLE";
                break;
            case View.GONE:
                name = "GONE";
                break;
            default:
                name = "UNKNOWN";
                break;
        }
        return name + '(' + visibility + ')';
    }

    private FooViewUtils()
    {
    }
}

File: android/smartfoo-android-lib-core/src/main/java/com/smartfoo/android/core/view/FooViewUtils.kt
package com.smartfoo.android.core.view

import android.view.View

/** View utility functions. */
object FooViewUtils {
    /**
     * Returns a human-readable label for a [android.view.View] visibility constant.
     *
     * @param visibility one of [android.view.View.VISIBLE], [android.view.View.INVISIBLE],
     *   or [android.view.View.GONE]
     * @return a descriptive name including the numeric value, e.g. `"VISIBLE(0))"`
     */
    fun viewVisibilityToString(visibility: Int): String {
        return when (visibility) {
            View.VISIBLE -> "VISIBLE"
            View.INVISIBLE -> "INVISIBLE"
            View.GONE -> "GONE"
            else -> "UNKNOWN"
        }.let { "$it($visibility))" }
    }
}
